src/XR/features/WebXRLightEstimation.ts

```diff
@@ -0,0 +1,161 @@
+     * Since creating and processing the cube map is potentially expensive, especially if mip maps are needed, you can listen to the onReflectionCubeMapUpdatedObservable to determine
+     * when it has been updated.
+     */
+    public get reflectionCubeMap(): Nullable<WebGLTexture> {
+        return this._reflectionCubeMap;
+    }
+
+    /**
+     * The XRLightProbe object created during attach (may not be available for a few frames depending on device).
+     *
+     * The XRLightProbe itself contains no lighting values, but is used to retrieve the current lighting state with each XRFrame.
+     * XR Light Probe also contains the probe space, may update it's pose over time as the user moves around their environment.
+     */
+    public get xrLightProbe(): Nullable<XRLightProbe> {
+        return this._xrLightProbe;
+    }
+
+    /**
+     * The most recent light estimate.  Available starting on the first frame where the device provides a light probe.
+     */
+    public get xrLightingEstimate(): Nullable<XRLightEstimate> {
+        return this._xrLightEstimate;
+    }
+
+    private _getXRGLBinding(): XRWebGLBinding {
+        if (this._xrWebGLBinding === null) {
+            // TODO: find a way that looks less hacky to get webgl context.  try webgl2 and fallback to webgl.
+            const canvas: HTMLCanvasElement = this._xrSessionManager.scene.getEngine().getRenderingCanvas()!;
+            let context: Nullable<WebGLRenderingContext | WebGL2RenderingContext> = canvas.getContext("webgl2");
+            if (!context) {
+                context = canvas.getContext("webgl")!;
+            }
+            this._xrWebGLBinding = new XRWebGLBinding(this._xrSessionManager.session, context);
+        }
+        return this._xrWebGLBinding;
+    }
+
+    /**
+     * Event Listener to for "reflectionchange" events.
+     */
+    private _updateReflectionCubeMap = (): void => {
+        this._reflectionCubeMap = this._getXRGLBinding().getReflectionCubeMap(this._xrLightProbe!);
+        this.onReflectionCubeMapUpdatedObservable.notifyObservers(this._reflectionCubeMap);
+    }
+
+    /**
+     * attach this feature
+     * Will usually be called by the features manager
+     *
+     * @returns true if successful.
+     */
+    public attach(): boolean {
+        if (!super.attach()) {
+            return false;
+        }
+
+        this._xrSessionManager.session.requestLightProbe({
+            reflectionFormat: this.options.reflectionFormatSRGBA8 ? "srgba8" : this._xrSessionManager.session.preferredReflectionFormat ?? "srgba8"
+        }).then((value: XRLightProbe) => {
+            this._xrLightProbe = value;
+            this._xrLightProbe.addEventListener('reflectionchange', this._updateReflectionCubeMap);
+            this._reflectionCubeMap = this._getXRGLBinding().getReflectionCubeMap(this._xrLightProbe!);

```

Comment:
should probably notify here (the observable)

Reply:
fixed.